### PR TITLE
credentials/alts: Properly release server InBytes buffer after the handshake is complete.

### PR DESCRIPTION
**Pull request by cesarghali**
Part of the the server-side `InBytes` buffer is eventually passed to `conn.NewConn()`. Since the connection resulting from calling that API stays open as long as the gRPC channel is open, the original 64KB `InBytes` buffer will stay references for the entire channel lifetime and will never be released. This PR fixes that issue.

Another attempt to fix the buffer ownership issue was done in #3513 but that got reverted in #3528.

### CODE REVIEW

File: credentials/alts/internal/conn/record.go [hunk 1]
```diff
@@ -120,16 +121,18 @@ func NewConn(c net.Conn, side core.Side, recordProtocol string, key []byte, prot
 		// altsRecordDefaultLength (bytes) data into protected at one
 		// time. Therefore, 2*altsRecordDefaultLength-1 is large enough
 		// to buffer data read from the network.
-		protected = make([]byte, 0, 2*altsRecordDefaultLength-1)
+		protectedBuf = make([]byte, 0, 2*altsRecordDefaultLength-1)
+	} else {
+		protectedBuf = make([]byte, len(protected))
```
Comment (menghanl) [inline credentials/alts/internal/conn/record.go:126]: This doesn't copy protected.

Is there an easy way to add a unit test?

Comment (cesarghali) [reply credentials/alts/internal/conn/record.go:126]: Not sure how I missed copying. This is the main reason behind this PR :)

We need a unit test for the case where `protected` is not nil. I'm working on adding that but it would require some setting up. We can add it in this PR or in a follow up one, either is fine with me.

Comment (cesarghali) [reply credentials/alts/internal/conn/record.go:126]: Done. I added a test with a non-nil protected buffer. PTAL.
